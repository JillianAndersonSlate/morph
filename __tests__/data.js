import { morph } from '../index.js'
import { join } from 'path'
import { readdirSync, readFileSync } from 'fs'

const as = 'data'
const isData = f => /\.data$/.test(f)
const getPath = (f = '.') => join(__dirname, 'data', f)
describe(as, () => {
  readdirSync(getPath()).filter(isData).forEach(f => {
    const name = f.replace(/\.view$/, '')
    const code = readFileSync(getPath(f), 'utf-8')

    it(`parses ${as} ${name}`, () => {
      expect(morph(code, { as, name, pretty: true })).toMatchSnapshot()
    })
  })
})
