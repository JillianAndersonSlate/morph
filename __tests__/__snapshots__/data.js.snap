// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`data parses data fields.data 1`] = `
Object {
  "code": "export default {
  city: 'Dublin',
  country: 'Ireland',
  fonts: {},
  props: {},
  todos: [],
};
",
}
`;

exports[`data parses data list.data 1`] = `
Object {
  "code": "export default [
  { name: 'Dario', surname: 'Cravero' },
  { name: 'Tom', surname: 'Parandyk' },
];
",
}
`;
